packages/interface-ipfs-core/src/miscellaneous/resolve.js

```diff
@@ -94,8 +94,23 @@ module.exports = (common, options) => {
       await ipfs.name.publish(path, { allowOffline: true })
       await ipfs.name.publish(`/ipns/${ipfs.peerId.id}`, { allowOffline: true, key: 'key-name', resolve: false })
 
-      return expect(await ipfs.resolve(`/ipns/${keyId}`, { recursive: true }))
+      return expect(await ipfs.resolve(`/ipns/${keyId}`))
         .to.eq(`/ipfs/${path}`)
     })
+
+    it('should resolve IPNS link non-recursively if recursive==false', async function () {
+      this.timeout(20 * 1000)
+      // webworkers are not dialable because webrtc is not available
+      const node = (await common.spawn({ type: isWebWorker ? 'go' : undefined })).api
+      await ipfs.swarm.connect(node.peerId.addresses[0])
+      const { path } = await ipfs.add(uint8ArrayFromString('should resolve an IPNS key if recursive === false'))
+      const { id: keyId } = await ipfs.key.gen('key-name', { type: 'rsa', size: 2048 })
```

Comment:
This test fails in CI because the key name is not unique:

```suggestion
      const { id: keyId } = await ipfs.key.gen(`key-name-${Math.random()}`, { type: 'rsa', size: 2048 })
```

```
ipfs:   1) interface-ipfs-core tests
ipfs:        .resolve
ipfs:          should resolve IPNS link non-recursively if recursive==false:
ipfs:      Error: Key 'key-name' already exists
ipfs:       at Keychain.createKey (file:///home/travis/build/ipfs/js-ipfs/node_modules/libp2p/src/keychain/index.js:222:45)
ipfs:       at Context2.<anonymous> (file:///home/travis/build/ipfs/js-ipfs/packages/interface-ipfs-core/src/miscellaneous/resolve.js:107:29)
ipfs: 
```